interpreter/interface.go

```diff
@@ -19,9 +19,12 @@
 package interpreter
```

Comment:
Should the changes to the `interpreter` package maybe also be applied on `master`?

Reply:
I was thinking about this. Since the refactoring needed changes from both interpreter and vm side, it was easier to do the refactor on the compiler branch (same for follow-up PRs). What I thought was to once the refactoring is done, get the diff of all the interpreter/stdlib packages and apply it on master.

Reply:
That's a good idea! Yeah, it might be good to just port the changes in a single PR 👍 We can just create a patch with all commits of the compiler branch, and filter to only include changed to the interpreter package